packages/grid/x-data-grid/src/hooks/features/density/useGridDensity.tsx

```diff
@@ -14,31 +14,22 @@ export const COMPACT_DENSITY_FACTOR = 0.7;
 export const COMFORTABLE_DENSITY_FACTOR = 1.3;
 
 // TODO v6: revise keeping headerHeight and rowHeight in state
-const getUpdatedDensityState = (
-  newDensity: GridDensity,
-  newHeaderHeight: number,
-  newRowHeight: number,
-): GridDensityState => {
+// Math.floor(height * COMPACT_DENSITY_FACTOR)
+const getUpdatedDensityState = (newDensity: GridDensity): GridDensityState => {
```

Comment:
While we are at it, can we get rid of `GridDensityTypes` completely and use `GridDensity` instead?
Then we can replace `getUpdatedDensityState` function with a simple lookup: 
```ts
const DENSITY_FACTORS: Record<GridDensity, number> = {
  compact: COMPACT_DENSITY_FACTOR,
  comfortable: COMFORTABLE_DENSITY_FACTOR,
  standard: 1,
};
```

Reply:
I'm not sure about this. For me, the current setup is more readable but then again this is a personal opinion. Also the change you are proposing changes the state structure as far as I see. I would like to keep it as is right now (at least for now).

Reply:
> Also the change you are proposing changes the state structure as far as I see.

No, I don't think so. It's just to simplify things since it feels a bit verbose to have this:

```tsx
export type GridDensity = 'compact' | 'standard' | 'comfortable';

enum GridDensityTypes {
  Compact = 'compact',
  Standard = 'standard',
  Comfortable = 'comfortable',
}
```

I can submit a PR later, this should remove some amount of code and we can discuss it there if you don't mind